libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -55,6 +55,7 @@ bool GCS_MAVLINK::ftp_init(void) {
     ftp.requests = nullptr;
     delete ftp.replies;
     ftp.replies = nullptr;
+    gcs().send_text(MAV_SEVERITY_WARNING, "failed to initialize MAVFTP; insufficient free RAM");
```

Comment:
This is not accurate - there are other things up there that already can cause this to be wrong.

I contend that if we fail to allocate something we should fail a prearm check - don't allow the user to fly in that case.

I *think* it will be a call in `AP_Arming/AP_Arming.cpp` to this very function - if it returns false then we fail the prearm and just say, `failed to initialize MAVFTP`

Reply:
but is failing to initialize MAVFTP something that should prevent flight? other things yes, but MAVFTP is not a required element for flight is it? I have been flying recently without the MAVFTP ability since I did not know that it was a RAM issue...turns out I had buffers allocated for FFT that I was not using, that fixed the MAVFTP problem and my Compass Calibration not starting issues...

Reply:
@peterbarker I could qualify "insufficient" with "almost certainly" :)

Reply:
ftp_init is not called until a GCS requests a file transfer. Since this can happen in flight and it **news**s some large buffers it seems pretty unsafe, so I think you're correct that it should be called at the prearm stage. Especially since those buffers are never freed.